addons/web/static/src/js/views/pivot/pivot_view.js

```diff
@@ -53,6 +53,10 @@ odoo.define('web.PivotView', function (require) {
             //Compute the measures and the groupableFields
             Object.keys(this.fields).forEach(name => {
                 const field = this.fields[name];
+                const measureField = this.arch.children.find(f => f.attrs.name === name);
+                if (measureField && measureField.attrs.invisible && py.eval(measureField.attrs.invisible)) {
+                    return;
+                }
```

Comment:
For consistency, I would do that here below where we remove the fiels when invisible.

```js
// Remove invisible fields from the measures
if (field.attrs.invisible && py.eval(field.attrs.invisible)) {
    delete measures[name];
    if (name in groupableFields) {
        delete groupableFields[name];
    }
    return;
}
```